internal/pkg/service/stream/definition/repository/source_repo.go

```diff
@@ -126,16 +126,20 @@ func (r *SourceRepository) Create(versionDescription string, input *definition.S
 }
 
 //nolint:dupl // similar code is in SinkRepository
-func (r *SourceRepository) Update(k key.SourceKey, versionDescription string, updateFn func(definition.Source) definition.Source) *op.AtomicOp[definition.Source] {
+func (r *SourceRepository) Update(k key.SourceKey, versionDescription string, updateFn func(definition.Source) (definition.Source, error)) *op.AtomicOp[definition.Source] {
 	var result definition.Source
```

Comment:
Here is the `SourceRepository`, the `Update` method, with the `updateFn` callback. 

Reminder: update is implemented via the callback, so operation can be retried on a collision and result is consistent.

Reply:
I updated the callback type, so it can return an error.